redis/src/aio/connection.rs

```diff
@@ -102,7 +117,9 @@ where
 
     /// Fetches a single response from the connection.
     async fn read_response(&mut self) -> RedisResult<Value> {
-        crate::parser::parse_redis_value_async(&mut self.decoder, &mut self.con).await
+        let result = crate::parser::parse_redis_value_async(&mut self.decoder, &mut self.con).await;
+        self.push_manager.try_send(&result, &self.con_addr);
```

Comment:
Currently it's not informed on disconnections since it doesn't do anything with it right now.
Also I didn't understand what you mean by `wouldn't it make more sense that the result will either be sent on the push manager or returned, but not both?`. It's kinda a hook for PushManager and it shouldn't change any behaviour.

Reply:
> Currently it's not informed on disconnections since it doesn't do anything with it right now.

So, someone who only holds the receiver end on a push manager won't know that the receiver is essentially dead? That seems problematic IMO. We should either disconnect the receiver, or send on it a disconnect notice, so that the user will know.

> It's kinda a hook for PushManager and it shouldn't change any behaviour.

fair enough

Reply:
Hmmm yes, I think notice on disconnection should be done, I will work on it.

Reply:
In clusted mode one PushManager is used by multiple connection, in a disconnection event of one connection shouldn't close the channel for all of them. So I think giving information about disconnection to the channel is better. Maybe a Result<PushInfo> would be enough ?

Reply:
You're right, cluster push managers are more complex. If we pass a `Result`, it would be hard for the user to understand which slots/keys are affected by the disconnect. And regardless, even if we passed some disconnect notification, because the cluster connection reconnects automatically, the user will need to understand when/how to reconnect. AFAIS in order to return an informative disconnect from a cluster, we need to pass a message with which subscription disconnected, and what type of subscriptions they were. Am I missing something?

Maybe we should split this, and only implement this for sync Connection and MultiplexedConnection (not for the traits), and implement the complex cases (`ConnectionManager`, sync and async cluster connections) each in separate PRs? Maybe with different push managers for each scenario. Or do you have a simple solution that works also for the complex cases?

Reply:
On second thought, as long as the user is informed about disconnects, `ConnectionManager` isn't a complex case, right?

Reply:
@nihohit If we are not going to use re-connection (which u didn't like because it made things complex), then only sending connection disconnection info is easier. PushInfo has `con_addr` it can be used for distinguishing connections, `PushKind::Other("Disconnection :X")` can be used or something like `PushKind::Disconnection` can be added.

Reply:
>  PushInfo has con_addr it can be used for distinguishing connections,

True, but there's no way for the user to know which address contains which slots, and so, which specific subscription was disconnected.

Reply:
It's possible to add hashslot to PushInfo which is u16 afaik. Adding disconnected subscriptions can be made with another struct.
PushKind::Disconnection(DisconnectionInfo) would be easier.

Reply:
> It's possible to add hashslot to PushInfo which is u16 afaik.

True, but each connection manages multiple slots, and ATM there's no convenient way of getting all of the slots for a given address - the slotmap is a map from slots to addresses, the other direction isn't available. 

> Adding disconnected subscriptions can be made with another struct.

Without tracking subscriptions (like you've did before), how would you know which subscriptions were disconnected for each dropped connection? And again, since this is relevant only for clusters, I would think that cluster connections might have a different push manager than single node connections.

Reply:
I will research more on this subject. Can we skip disconnection/reconnection handling in this pr ?

Reply:
IMO we can skip reconnections, but I don't see how we can skip disconnection - that will leave an API gap, since the users won't know when their pub/sub is not working.
What we can skip, IMO, is cluster pub/sub, since disconnections on single connections are much simpler - everything disconnects together.

Reply:
@nihohit okey is it okay if we leave cluster pub/sub as it is but adding disconnection information on single connections. It's possible to use `PushKind::Disconnection(DisconnectionInfo) ` for single disconnections then I can expand it with cluster information in later PRs.

Reply:
> is it okay if we leave cluster pub/sub as it is

I prefer not to keep code that we consider to be incorrect - at the very least, it should be very clearly documented as partial, or better yet - fail in an informative manner. But IMO the best solution is just to keep the current state of the code in a separate branch, and continue the research / design discussion on the separate branch, while a more minimal branch gets merged.

Reply:
I've added `Disconnection` to PushKind and handled things in cm,mpx,sync. I didn't implemented in aio since it's deprecated.